modeshape-jcr/src/main/java/org/modeshape/jcr/bus/RepositoryChangeBus.java

```diff
@@ -76,7 +101,11 @@ public boolean register( ChangeSetListener observer ) {
         if (observer == null || shutdown.get()) return false;
         try {
             registrationLock.lock();
-            return ringBuffer.addConsumer(observer);
+            boolean result = ringBuffer.addConsumer(observer);
+            if (result && statistics != null) {
+                statistics.increment(ValueMetric.LISTENER_COUNT);               
+            }      
+            return result;
```

Comment:
It probably would be better to use a no-op `RepositoryStatistics` so that `statistics` is never null. It'd simplify the logic in these calls a bit, plus allow the `statistics` field to be `final` (allow for some additional JIT optimizations).